### ISSUE TITLE
function copy

### ISSUE THREAD

**Issue by Oxicode** [thecodingmachine/safe#25]
please

**Comment by mallardduck** [comment]
Not on the project, but I'm curious what you'd like to see done to PHP's `copy`?
It's already a consistent function that simply returns a boolean no mater what.
Does this type of function really need to be wrapped too?

**Comment by mallardduck** [comment]
@moufmouf Absolutely! I can totally agree with your sentiments. I think I just didn't explain my thought very well. Essentially I feel that the return value is less important now that a failure status is indicated by an Exception.

I know that's still a pretty poor explanation, but IMO there's no need to capture the result of this type of function when it's always returning true. Instead you can just proceed as if you assume a success all the time and everything will break gracefully on the exception. If you need to catch the exception, then do that - but you no longer really need the return value at that point.

In the end these are things that I can see leading to cleaner looking code overall. So that's a positive for sure.

**Comment by moufmouf** [comment]
> If you need to catch the exception, then do that - but you no longer really need the return value at that point.

Absolutely! If you have a look at the generated function in "safe", you will see that for the functions that the functions that used to return a boolean for success/failure are now returning nothing.

For instance, chown used to return a bool, but in Safe, it returns nothing (void): https://github.com/thecodingmachine/safe/blob/50ea87162e0db259391960b92f5a057c4037fcd8/generated/filesystem.php#L81
